doc/howtos/firewall-configuration.rst

```diff
@@ -119,3 +119,11 @@ Opening holes to firewall using this method is not absolutely mandatory.
 If you are building a device without dynamic services or applications,
 you can simply add the necessary rules to the custom ruleset that is
 loaded upon device startup.
+
+Note that if you are starting your service as non-root user (as you
+probably should be) using systemd's ``User=`` option, you'll need to
+make sure that the firewall manipulation commands are run with original
+permissions. An easy way to do that is to use
+``PermissionsStartOnly=true`` option in the service file. This works if
+you don't need to run any other ``ExecStartPre=`` or similar commands
+with the reduced non-root privileges.
```

Comment:
Yes, they are generated, but with a twist...

AppFW has to actually do this in a more complex way due to the fact that the `ExecStartPre=` command and similar declared in manifest have to be run with the same permissions as the app itself. Thus the firewall commands, which need to be run as root, have to be started from a different sub-service that the actual application service depends on. @klihub already has this working.